util/options.py

```diff
@@ -258,26 +261,35 @@ def try_read_experiments_from_images(
         for arg in args:
             # Don't expand wildcards if URI-style filename
             if "*" in arg and not get_url_scheme(arg):
-                args_new.extend(glob(arg))
+                filenames = glob(arg)
+                if not filenames:
+                    logger.warning("No files found with pattern %s" % arg)
```

Comment:
Suggest f-string you always want to expand, logger comma notation if not - there shouldn't be a need to ever use `%` in the context of logging.

This aside - shouldn't `arg` end up in `unhandled`?
Should this only be a warning or should it be a `sys.exit(...)`?

Reply:
I thought logger was an exception to using f-strings where it's [optimised](https://docs.python.org/3/howto/logging.html#optimization) for `%` style formatting.

I didn't use `sys.exit(...)` to allow the help to be called downstream, which is consistent with what happens when you try and open files from a single directory with a wild card.

Reply:
To do this "properly" with logging it should be:
`logger.warning("No files found with pattern %s", arg)`
not
`logger.warning("No files found with pattern %s" % arg)`
i.e. you pass `arg` as an argument to the `logger.warning` function rather than using `%` string formatting.
See also https://docs.python.org/3/library/logging.html#logging.Logger.debug

This is only really advantage if `arg` is an object whose string representation is evaluated lazily (and relatively expensive). Otherwise you are probably best just using f-Strings now, i.e.
`logger.warning(f"No files found with pattern {arg}")`
(although opinions here may vary)

Reply:
Re: `arg` going into `unhandled`, I did I try this originally. But unhandled arguments here currently get raised at the options level, which just gives a `Sorry: Unable to handle the following arguments` with no explanation, which doesn't seem particularly helpful to the user. The IO should be being handled at the import (top) level really, but I assumed changing the behaviour of `parse_args` in `options` was a bit risky with the current quick_parse/ignore_unhandled/return_unhandled mix of behaviours...

Reply:
As a minor note on this (admittedly old) conversation, logger is an exception where the logger-style formatting should _in theory_ be used, but I've long given up trying to force this since using f-strings in loggers is so prevalent.